### PR TITLE
Preview objects that are Dragged and dropped in the navbar to be deleted

### PR DESCRIPTION
**Pull request by alstar555**
Added a preview for objects that are dragged and dropped into the navigation bar to be deleted
fixedIssueNavPreview
.

### CODE REVIEW

File: src/site/shared/containers/ItemNav/index.tsx [hunk 1]
```diff
@@ -69,6 +71,9 @@ export const ItemNav = <D,>({ info, config, additionalData,
 
     const [hovering, setHover] = useState("");
 
+    //track whether mouse is over Nav bar
```
Comment (TGCrystal) [inline src/site/shared/containers/ItemNav/index.tsx:74]: nit: add space after `//`

File: src/site/shared/containers/ItemNav/index.tsx [hunk 2]
```diff
@@ -93,6 +98,39 @@ export const ItemNav = <D,>({ info, config, additionalData,
...
+            return null;
+        if (!(currentlyPressedObj instanceof Component))
+            return null;
+        if(!overNav){
+            return null;
+        }
+        const id = GetIDFor(currentlyPressedObj);
+        var section_id;
+        var config_imgRoot;
+        var item_icon;
```
Comment (TGCrystal) [inline src/site/shared/containers/ItemNav/index.tsx:112]: nit: use `let` instead of `var`

File: src/site/shared/containers/ItemNav/index.tsx [hunk 3]
```diff
@@ -93,6 +98,39 @@ export const ItemNav = <D,>({ info, config, additionalData,
         // Else just delete
         info.history.add(CreateDeleteGroupAction(info.designer, [currentlyPressedObj]).execute());
     }
+    function deletingItemPreview(){
+        if (!currentlyPressedObj)
+            return null;
+        if (!(currentlyPressedObj instanceof Component))
+            return null;
+        if(!overNav){
```
Comment (TGCrystal) [inline src/site/shared/containers/ItemNav/index.tsx:106]: nit: add space between `if` and `(`

File: src/site/shared/containers/ItemNav/index.tsx [hunk 4]
```diff
@@ -183,12 +221,13 @@ export const ItemNav = <D,>({ info, config, additionalData,
...
         <div className="itemnav__preview"
              style={{
-                display: (curItemImg ? "initial" : "none"),
-                left: pos.x,
-                top: pos.y,
-             }}>
+                 display: (curItemImg ? "initial" : "none"),
+                 left: pos.x,
+                 top: pos.y,
+                }}>
```
Comment (TGCrystal) [inline src/site/shared/containers/ItemNav/index.tsx:230]: Not sure if linting said to change this spacing, but if it didn't then it should be what it was before

**Review by TGCrystal — CHANGES_REQUESTED** [review]
Functionality looks good to me. It doesn't work on mobile, but neither does dragging to the itemnav to delete at all. Fixing that would probably be outside the scope of this pr, so you don't have to worry about it.

There is probably some way that the general logic should be refactored instead of relying on a large nested function. I'm not sure what exactly that is, so you should consult @LeonMontealegre on that.

**Review by LeonMontealegre — DISMISSED** [review]

**Pull request merged**